### PR TITLE
influxdb2-server: 2.1.1 -> 2.4.0

### PR DESCRIPTION
**Pull request by danderson**
influxdb2-cli: 2.3.0 -> 2.4.0

###### Description of changes

Update to latest upstream release. Release notes don't suggest any breaking changes or required manual upgrade steps: https://docs.influxdata.com/influxdb/v2.4/reference/release-notes/influxdb/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by danderson** [comment]
Result of `nixpkgs-review pr 191753` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>influxdb2</li>
    <li>influxdb2-cli</li>
    <li>influxdb2-server</li>
  </ul>
</details>

**Comment by danderson** [comment]
CI failure is spurious, evaluator ran out of disk space.

**Comment by danderson** [comment]
Removed Valentin from reviewer list, I think he got added because of a mistaken docs change. The change doesn't have doc changes any more.

File: pkgs/servers/nosql/influxdb2/default.nix [hunk 1]
```diff
@@ -81,7 +84,7 @@ in buildGoModule {
         exit 1
       fi
 
-      ui_ver=$(grep influxdata/ui/releases scripts/fetch-ui-assets.sh | ${perl}/bin/perl -pe 's#.*/OSS-([^/]+)/.*#$1#')
+      ui_ver=$(egrep 'influxdata/ui/releases/.*/sha256.txt' scripts/fetch-ui-assets.sh | ${perl}/bin/perl -pe 's#.*/OSS-([^/]+)/.*#$1#')
```
Comment (risicle) [inline pkgs/servers/nosql/influxdb2/default.nix:87]: `egrep` is being deprecated. Prefer `grep -E`.

**Comment by risicle** [comment]
@ofborg test influxdb cadvisor prometheus-exporters.influxdb

Could do with some of these in `passthru.tests`

**Comment by risicle** [comment]
Otherwise, `nixpkgs-review` happy, macos 10.15 & nixos x86_64

**Pull request merged**